Emby.Server.Implementations/EntryPoints/UdpServerEntryPoint.cs

```diff
@@ -62,7 +62,7 @@ public void Dispose()
             }
 
             _cancellationTokenSource.Cancel();
```

Comment:
The async function should stop since the task is canceled

Reply:
It's erroring on line 104

var result = await _udpSocket.ReceiveFromAsync(_receiveBuffer, SocketFlags.None, _endpoint).ConfigureAwait(false);

before the exception can be thrown. The dispose has whilst this function is running. 

The error is a null pointer and _udpSocket is null.

As udpserver will get disposed in the normal way, i can't see harm in letting the system dispose of it.

Reply:
System.ObjectDisposedException
  HResult=0x80131622
  Message=Cannot access a disposed object.
Object name: 'System.Net.Sockets.Socket'.
  Source=System.Net.Sockets
  StackTrace:
   at System.Net.Sockets.Socket.EndReceiveFrom(IAsyncResult asyncResult, EndPoint& endPoint)
   at System.Net.Sockets.Socket.<>c.<ReceiveFromAsync>b__283_0(IAsyncResult iar)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Emby.Server.Implementations.Udp.UdpServer.<BeginReceiveAsync>d__11.MoveNext() in C:\code\Jellyfin\Emby.Server.Implementations\Udp\UdpServer.cs:line 106

  This exception was originally thrown at this call stack:
    [External Code]
    Emby.Server.Implementations.Udp.UdpServer.BeginReceiveAsync(System.Threading.CancellationToken) in UdpServer.cs

Reply:
Changed it from being run by a task.run to an async and this issue has gone away!

Reply:
Put the dispose back into the Net/Udp and it errors with

System.ObjectDisposedException
  HResult=0x80131622
  Message=Cannot access a disposed object.
Object name: 'System.Net.Sockets.Socket'.
  Source=System.Net.Sockets
  StackTrace:
   at System.Net.Sockets.Socket.EndReceiveFrom(IAsyncResult asyncResult, EndPoint& endPoint)
   at Emby.Server.Implementations.Net.UdpSocket.EndReceive(IAsyncResult result) in C:\code\Jellyfin\Emby.Server.Implementations\Net\UdpSocket.cs:line 137

Reply:
So this leaves 3 options

1) leave as is and ignore the exception (it's in a try/catch)

2) don't dispose the objects. Leave that to the GC

3) put in locking on the dispose to wait for the completion of the tasks.